src/gpuocean/drifters/MLDrifterCollection.py

```diff
@@ -0,0 +1,175 @@
+# -*- coding: utf-8 -*-
+
+"""
+This software is a part of GPU Ocean.
+
+Copyright (C) 2018, 2023, 2024  SINTEF Digital
+
+This python class implements a DrifterCollection living on the CPU.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+
+import numpy as np
+
+from gpuocean.utils import Common
+from gpuocean.drifters import CPUDrifterCollection
+
+class MLDrifterCollection(CPUDrifterCollection.CPUDrifterCollection):
+    """
+    Class holding a collection of drifters for a 
+    """ 
+    def __init__(self, numDrifters, ensemble_size, observation_variance=0.01,
+                 boundaryConditions=Common.BoundaryConditions(), 
+                 initialization_cov_drifters=None,
+                 domain_size_x=1.0, domain_size_y=1.0,
+                 fixed_random_walk_direction=True):
+        """
+        Creates a collection of drifters suitable for multi-level (ML) ensembles.
+
+        Most relevant parameters
+        numDrifters: number of drifters represented in the collection
+        ensemble_size: number of realization per drifter. 
+        boundaryConditions: BoundaryConditions object, relevant during drift
+        domain_size_{x,y}: size of computational domain in meters
+        """
```

Comment:
update doc slightly... Add a TODO to rename to StatisticalDrifterCollection (but not do it since we otherwise break multi-level code...)